controllers/grid/admin/languages/PKPAdminLanguageGridHandler.inc.php

```diff
@@ -330,30 +329,31 @@ function enableLocale($args, $request) {
 	 * @return JSONMessage JSON object
 	 */
 	function disableLocale($args, $request) {
-		$rowId = $request->getUserVar('rowId');
+		$locale = $request->getUserVar('rowId');
 		$gridData = $this->getGridDataElements($request);
 		$notificationManager = new NotificationManager();
 		$user = $request->getUser();
 
-		if (array_key_exists($rowId, $gridData)) {
+		if ($request->checkCSRF() && array_key_exists($locale, $gridData)) {
 			// Don't disable primary locales.
-			if ($gridData[$rowId]['primary']) {
+			if ($gridData[$locale]['primary']) {
 				$notificationManager->createTrivialNotification(
 					$user->getId(), NOTIFICATION_TYPE_ERROR,
 					array('contents' => __('admin.languages.cantDisable'))
 				);
-			} else {
-				$locale = $rowId;
-				$this->_updateLocaleSupportState($request, $rowId, false);
-
-				$notificationManager->createTrivialNotification(
-					$user->getId(), NOTIFICATION_TYPE_SUCCESS,
-					array('contents' => __('notification.localeDisabled'))
-				);
+				return new JSONMessage(false);
```

Comment:
In order for the notification ('admin.languages.cantDisable') to be displayed this should be called here, I believe: `return DAO::getDataChangedEvent($locale);` -- as it was earlier. Else the confirmation modal just stays open and nothing happens...